lms/services/blackboard_api/_basic.py

```diff
@@ -70,6 +70,31 @@ def get_token(self, authorization_code):
             authorization_code=authorization_code,
         )
 
+    def refresh_token(self):
+        """
+        Make a refresh token request and save the new token in the DB.
+
+        :raise OAuth2TokenError: if we don't have a refresh token for the user
+            or if our refresh token is invalid (e.g. it's expired or been
+            revoked)
+        :raise HTTPError: if the HTTP request fails
+        :raise ValidationError: if the server's access token response is invalid
+        """
+        try:
+            self._oauth_http_service.refresh_token_request(
+                token_url=self.token_url,
+                redirect_uri=self.redirect_uri,
+                auth=(self.client_id, self.client_secret),
+            )
+        except HTTPError as err:
+            error_dict = BlackboardErrorResponseSchema(err.response).parse()
+
+            if error_dict.get("error") == "invalid_grant":
+                # Looks like our refresh token has expired or been revoked.
+                raise OAuth2TokenError() from err
+
+            raise
```

Comment:
This error handling could actually move down into `OAuthHTTPService`, it's standard OAuth stuff: https://datatracker.ietf.org/doc/html/rfc6749#section-5.2